src/plugins/vis_augmenter/common/types.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ExpressionFunctionDefinition } from '../../expressions';
+
+export interface VisLayer {
+  // will be used as the column ID
+  id: string;
+  // will be used as the name when hovering over the tooltip
+  name: string;
+}
+
+export type VisLayers = VisLayer[];
+
+// resourceId & resourceName are required so that the
+// events flyout can partition data based on these attributes
+// (e.g., partitioning anomalies based on the detector they came from)
+export interface PointInTimeEventMetadata {
+  resourceId: string;
+  resourceName: string;
+  tooltip?: string;
```

Comment:
I think we'll eventually want more structure and control here. What about modeling it as some sort of callback or string? Or do you think it would be better to keep it simple for now?

Reply:
Yeah good callout, I set it to `string` as kind of a placeholder for now. I need to dig into #3145 more to get a better idea of what's a useful way to propagate this. If it makes sense to you, I'd like to leave as `string` for now and we can re-structure it if necessary down the road.